flake.nix

```diff
@@ -6,46 +6,69 @@
       url = "github:oxalica/rust-overlay";
       inputs.nixpkgs.follows = "nixpkgs";
     };
-    naersk.url = "github:nix-community/naersk";
   };
 
-  outputs = { self, nixpkgs, rust-overlay, naersk, flake-utils }:
+  outputs = { self, nixpkgs, rust-overlay, flake-utils }:
     let systems = [ "x86_64-linux" "aarch64-linux" ];
     in flake-utils.lib.eachSystem systems (system:
       let
         overlays = [ (import rust-overlay) ];
         pkgs = import nixpkgs { inherit system overlays; };
         lib = pkgs.lib;
 
-        naersk' = pkgs.callPackage naersk {
-          cargo = pkgs.rust-bin.stable.latest.default;
-          rustc = pkgs.rust-bin.stable.latest.default;
-        };
+        cargoToml = builtins.fromTOML (builtins.readFile ./Cargo.toml);
+        cargoPackageVersion = cargoToml.package.version;
 
-        buildXwaylandSatellite = { withSystemd ? false }: naersk'.buildPackage {
-          src = ./.;
+        commitHash = self.shortRev or self.dirtyShortRev or "unknown";
 
-          nativeBuildInputs = with pkgs; [
-            rustPlatform.bindgenHook
-            rust-bin.stable.latest.default
-            pkg-config
+        version = "${cargoPackageVersion}-unstable-${commitHash}";
```

Comment:
> Isn't this kind off wrong given that at some point, a commit will be tagged for release, and then suddenly the release tag's package is called unstable? i think assigning an unstable label in this repo feels weird.

I think you're right about making the change, but not for the reason you mentioned. Whether or not the commit is tagged for release, using 'unstable' would have provided a human-readable way to indicate it's built from the main branch. However, since the user controls the input URL and can select a specific release, branch, or commit, 'unstable' wouldn't make sense in those cases.

> it also works weirdly given that the build output is now dependent on the git state

That's a valid point, but I don't find it very restrictive. I'm perfectly ok with that.